vars/mtaBuild.groovy

```diff
@@ -61,10 +63,23 @@ void call(Map parameters = [:]) {
 
         dockerExecute(script: script, dockerImage: configuration.dockerImage, dockerOptions: configuration.dockerOptions) {
 
-            // Apply maven user-settings (for custom repositories, etc)
-            if (configuration.projectSettingsFile) {
+            def projectSettingsFile = configuration.projectSettingsFile
+            if (projectSettingsFile?.trim()) {
+                if(projectSettingsFile.trim().startsWith("http")){
+                    downloadSettingsFromUrl(projectSettingsFile)
+                    projectSettingsFile = "settings.xml"
+                }
                 sh 'mkdir -p $HOME/.m2'
-                sh "cp ${configuration.projectSettingsFile} \$HOME/.m2/settings.xml"
+                sh "cp ${projectSettingsFile} \$HOME/.m2/settings.xml"
+            }
+
+            def globalSettingsFile = configuration.globalSettingsFile
+            if (globalSettingsFile?.trim()) {
+                if(globalSettingsFile.trim().startsWith("http")){
+                    downloadSettingsFromUrl(globalSettingsFile)
+                    globalSettingsFile = "settings.xml"
+                }
+                sh "cp ${globalSettingsFile} \$M2_HOME/conf/settings.xml"
```

Comment:
This seems a bit redundant. The only difference between project and global settings is making the dir and where the file is copied in the end. Is there any way to export the method and combine this?

Reply:
I moved the download part into a helper method. I don't think there is benefit in moving the larger block as this would then need an arbitrary shell script as parameter, so not much gained here in terms of DRY